### DISCUSSION TITLE
Create an option for Cline to automatically or ask for permission to refactor code that is getting too long.

### DISCUSSION THREAD

**Discussion by joewilsonai** [cline/cline#1041] [Feature Requests]
I think a lot of new users to coding don't realize that refactoring will help with managment of code but also make the LLMs more efficient.

**Comment by jeanesh** [comment]
Playing around with Pythagora, it seems to have this built in. It automatically refractors after every change and removes unnecessary code. I do find it Cline goes down the wrong path you end up with a lot of redundant code that has to be cleaned up after.
